uni/lib/view/schedule/schedule.dart

```diff
@@ -82,12 +91,15 @@ class SchedulePageViewState extends State<SchedulePageView>
         Expanded(
           child: TabBarView(
             controller: tabController,
-            children: Iterable<int>.generate(6).map((day) {
+            children: Iterable<DateTime>.generate(
+              6,
+              (int index) => currentWeek.getWeekday(index + 1),
+            ).map((day) {
               final lectures = lecturesOfDay(widget.lectures, day);
               if (lectures.isEmpty) {
-                return emptyDayColumn(context, day);
+                return emptyDayColumn(context, day.weekday - 1);
               } else {
-                return dayColumnBuilder(day, lectures, context);
+                return dayColumnBuilder(day.weekday - 1, lectures, context);
```

Comment:
Nit: shouldn't you be using the `WeekdayMapper` to translate between systems? I think it would be more semantic too

Reply:
I initially used the weekday mapper to do this, but it felt very verbose, so I removed it from here. I can do it anyways!

Reply:
Done, please tell me if you like how it turned out!